samples/subsys/logging/kernel_event_logger/src/main.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel_structs.h>
+#include <logging/kernel_event_logger.h>
+#include <kernel_structs.h>
+#include <misc/printk.h>
+#include <misc/util.h>
+#include <zephyr.h>
+
+K_THREAD_STACK_DEFINE(threadA_stack, 1024);
+K_THREAD_STACK_DEFINE(threadB_stack, 1024);
+K_THREAD_STACK_DEFINE(event_logger_stack, 4096);
+
+K_SEM_DEFINE(sem_sync, 0, 1);	/* starts off "not available" */
+
+static struct k_thread threadA_data;
+static struct k_thread threadB_data;
+static struct k_thread event_logger_thread_data;
+static u32_t timestamp;
+char thread_event_type[][12] = {"REDAY_Q", "PEND_STATE", "EXIT_STATE"};
+
+static void event_logger_thread(void)
```

Comment:
why not make this the main thread?

Reply:
The thread which reads event from the kernel event logger should be of least priority, because this thread should not affect application related threads. As main thread is set with priority zero, it should not be used for reading events from event loggers

Reply:
you can change the main thread priority using `CONFIG_MAIN_THREAD_PRIORITY`

Reply:
for example:  https://gist.github.com/nashif/372f216ca8fe84c560489c9171a9c219

Reply:
ok .